ruby/vampires.rb

```diff
@@ -0,0 +1,55 @@
+Vampire_Names = "dra drac ula cula fang nosf eratu"
+Allergies = %w(peanuts sunshine wheat beestings)
+Employee_Roster = []
+
+
+def vamp_namer(name)
+	vamp_array = Vampire_Names.split(" ")
+	vamp_array.each { |vamp_ele| return true if name.downcase.include?(vamp_ele) }
+	false
+end
+
+puts "How many employees to process?"
+num = gets.chomp.to_i
+
+num.times do
+	employee_allergies = Hash.new(0)
+
+	puts "What is your name?"
+	name = gets.chomp
+
+	puts "How old are you?"
+	age = gets.chomp.to_i
+
+	puts "What year were you born?"
+	birth_year = gets.chomp.to_i
+
+	puts "Our company cafeteria serves garlic bread. Should we order some for you? (y / n)"
+	garlic_order = gets.chomp
+
+	puts "Would you like to enroll in the company’s health insurance? (y / n)"
+	insurance = gets.chomp
+
+	Allergies.each do |allergy|
+		puts "Are you allergic to #{allergy}? (y / n)"
+		response = gets.chomp
+		employee_allergies[allergy] = response
+	end
+
+	case
+	when age < 116 && birth_year > 1900 && !vamp_namer(name)
+		puts "Probably not a vampire... Actually, never mind! What do these questions have to do with anything? Let's all be friends."
+	when (age > 116 || birth_year < 1900) && (garlic_order == "n" || insurance == "n")
+		puts "Probably a vampire... Actually, never mind! What do these questions have to do with anything? Let's all be friends."
+	when employee_allergies["sunshine"] == "y"
```

Comment:
'Match the following conditions in the order they're listed.' We first read this when starting off the challenge and we may try to hack it out with a basic if/elsif structure but we quickly learn that the first condition to pass will be the result even if a 'more true' version is below.  We may be tempted to switch the order for a working solution, but if we are to follow the constraints we need to be more creative. What if we made each individual scenario it's own respective if block? What would go inside each if block? Surely we can't just puts out the result otherwise our screen would be full of each vampire check that passed. What we could do is keep setting a variable result within each if block and then at the end of all the blocks display the result to look something like this
```
if condition_1
	result = 'low'
end

if condition_2
	result = 'medium'
end

if condition_3
	result = 'high'
end

puts result
```